src/Compilers/Core/Portable/InternalUtilities/JsonWriter.cs

```diff
@@ -146,13 +143,116 @@ private void WriteEnd(char c)
             _pending = s_newLine;
             _indent--;
             WritePending();
-            _outptut.Write(c);
+            _output.Write(c);
             _pending = s_commaNewLine;
         }
 
         public void Dispose()
         {
-            _outptut.Dispose();
+            _output.Dispose();
+        }
+
+        // String escaping implementation forked from System.Runtime.Serialization.Json to 
+        // avoid a large dependency graph for this small amount of code:
+        //
+        // https://github.com/dotnet/corefx/blob/master/src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/JavaScriptString.cs
+        //
+        // Possible future improvements: https://github.com/dotnet/roslyn/issues/9769
+        //
+        //   - Avoid intermediate StringBuilder and send escaped output directly to the destination.
+        //
+        //   - Stop escaping '/': it is is optional per JSON spec and several users have expressed
+        //     that they don't like the way '\/' looks.
+        //
+        private static string EscapeString(string value)
+        {
+            StringBuilder b = null;
+
+            if (string.IsNullOrEmpty(value))
+            {
+                return string.Empty;
+            }
+
+            int startIndex = 0;
+            int count = 0;
+            for (int i = 0; i < value.Length; i++)
+            {
+                char c = value[i];
+
+                if (c == '\"' || c == '\'' || c == '/' || c == '\\' || ShouldAppendAsUnicode(c))
+                {
+                    if (b == null)
+                    {
+                        b = new StringBuilder(value.Length + 5);
```

Comment:
Looks to be in Workspaces (therefore not accessible from here). Should I use PooledStringBuilder instead?